_invokers/java.md

```diff
@@ -9,46 +9,180 @@ href: https://github.com/projectriff/java-function-invoker
 
-If you provide a `main` parameter you need to include Spring Boot and
-all the other dependencies of the context in your archive. A Spring
-Boot jar file does not need an explicit `main` (there is one in the
-`MANIFEST.MF`), but if you supply one it will be used. If the
-`Function` has POJO (i.e. not generic JDK classes) as parameter types,
-then you also need to depend on `spring-cloud-function-context` (and
-include that in the archive).
+If you provide a `main` parameter you need to include Spring Boot and all the other dependencies of the context in your archive.
+A Spring Boot jar file does not need an explicit `main` (there is one in the `MANIFEST.MF`), but if you supply one it will be used.
+If the `Function` has POJO (i.e. not generic JDK classes) as parameter types, then you also need to depend on `spring-cloud-function-context` (and include that in the archive).
 
 Example:
 
 ```
---artifact greeter-1.0.0.jar --handler=functions.Greeter
+handler=functions.Greeter
 ```
 
 Example with Spring Boot application:
 
 ```
---artifact greeter-1.0.0.jar --handler=greeter
+handler=greeter
+```
+
+Example with Spring application context and an explicit main
+
+```
+handler='greeter&main=functions.FunctionApp'
+```
+
+#### Function samples
+
+We have sample application for [plain Java function](https://github.com/projectriff/java-function-invoker/tree/master/samples/greeter) and for [Spring Boot app with a function bean](https://github.com/projectriff/java-function-invoker/tree/master/samples/uppercase).
+
+### Cloud Native Buildpacks
+
+[Buildpacks](https://buildpacks.io/) provide a higher-level abstraction for building apps compared to Dockerfiles.
+
+The [riff project](https://github.com/projectriff/riff) provides its own [buildpack](https://github.com/projectriff/riff-buildpack) that specifically targets building functions using the riff function invokers.
+
+For the remainder of this document we will be using the riff buildpack to build and package our functions.
+
+### Building and deploying functions to local cluster
+
+To build and deploy your function locally you can use the `riff` CLI which can be installed following the instructions on the [riff Release page](https://github.com/projectriff/riff/releases).
+
+You also need to have riff and Knative installed on a local [Minikube](https://projectriff.io/docs/getting-started/minikube/) or [Docker for Mac](https://projectriff.io/docs/getting-started/docker-for-mac/) cluster, just follow the instructions we linked to.
+
+If you are using Minikube then you should configure Docker to use the Docker environment running in Minikube:
+
+```sh
+eval $(minikube docker-env)
+```
+
+Now you can build and deploy your function from the base directory of your app source. You need to provide the `--handler` option (see above for different handler types).
+To build a Boot app with a function bean, use:
+
+```sh
+riff function create java upper --handler upper --local-path . --image dev.local/upper:v1
```

Comment:
Should remove the `java` (invoker) arg here

Reply:
good catch